src/layouts/MainLayout.tsx

```diff
@@ -1,10 +1,29 @@
 import React from 'react';
 import BottomNav, { BottomNavHight } from '@/components/BottomNav';
 import styled from 'styled-components';
+import HomeHeader from '@/components/Home/HomeHeader';
+import Header from '@/components/Header';
+import { useSelector } from 'react-redux';
+import { RootState } from '@/store';
+import ExploreHeader from '@/components/Explore/ExploreHeader';
+import ProfileHeader from '@/components/Profile/ProfileHeader';
 
 const MainLayout = ({ children }: { children: React.ReactNode }) => {
+  const { status, current } = useSelector((state: RootState) => state.router);
+  const main = status === 'MAIN';
+  const home = current === 'HOME';
+  const explore = current === 'EXPLORE';
+  const profile = current === 'PROFILE';
```

Comment:
요런 분기점은 이렇게 객체로 관리해도 좋을 것 같습니다 😁
```javascript
 const Header = {
  HOME: <HomeHeader/>
  // ...
}
```
```suggestion
  const header = Header[current]
```

Reply:
네 좋아요 3개까지는 가독성이 괜찮아서 4개 넘어가면 수정해보도록 할게요!!